pandas/core/generic.py

```diff
@@ -7660,6 +7618,204 @@ def nanptp(values, axis=0, skipna=True):
+    2000-01-03   6.0   8.0  10.0
+    2000-01-04   NaN   NaN   NaN
+    2000-01-05   NaN   NaN   NaN
+    2000-01-06   NaN   NaN   NaN
+    2000-01-07   NaN   NaN   NaN
+    2000-01-08  21.0  23.0  25.0
+    2000-01-09  24.0  26.0  28.0
+    2000-01-10  27.0  29.0  31.0
+
+    >>> def log_T(s):
+    ...     return np.log(10 + s)
+    >>> def normalize_T(s):
+    ...     return (s - s.mean()) / s.std()
+
+    >>> dic = {
+    ...     'A': lambda x: x+1,
+    ...     'B': lambda x: x+2,
+    ... }
+    >>> df.transform(dic)
+               A     B
+    2000-01-01   1.0   3.0
+    2000-01-02   4.0   6.0
+    2000-01-03   7.0   9.0
+    2000-01-04   NaN   NaN
+    2000-01-05   NaN   NaN
+    2000-01-06   NaN   NaN
+    2000-01-07   NaN   NaN
+    2000-01-08  22.0  24.0
+    2000-01-09  25.0  27.0
+    2000-01-10  28.0  30.0
+
+    >>> dic = {'B': [log_T, normalize_T]}
+    >>> df.transform(dic)
+                    B
+                log_T normalize_T
+    2000-01-01  2.397895   -1.143001
+    2000-01-02  2.639057   -0.889001
+    2000-01-03  2.833213   -0.635001
+    2000-01-04       NaN         NaN
+    2000-01-05       NaN         NaN
+    2000-01-06       NaN         NaN
+    2000-01-07       NaN         NaN
+    2000-01-08  3.465736    0.635001
+    2000-01-09  3.555348    0.889001
+    2000-01-10  3.637586    1.143001
+
+    >>> df[['A', 'B']].transform([log_T, normalize_T])
+                    A                     B
+                log_T normalize_T     log_T normalize_T
+    2000-01-01  2.302585   -1.143001  2.397895   -1.143001
+    2000-01-02  2.564949   -0.889001  2.639057   -0.889001
+    2000-01-03  2.772589   -0.635001  2.833213   -0.635001
+    2000-01-04       NaN         NaN       NaN         NaN
+    2000-01-05       NaN         NaN       NaN         NaN
+    2000-01-06       NaN         NaN       NaN         NaN
+    2000-01-07       NaN         NaN       NaN         NaN
+    2000-01-08  3.433987    0.635001  3.465736    0.635001
+    2000-01-09  3.526361    0.889001  3.555348    0.889001
+    2000-01-10  3.610918    1.143001  3.637586    1.143001
+
+    >>> s = pd.Series(np.sin([3*x for x in range(10)]), name='A',
+    ...               index=pd.date_range('1/1/2000', periods=10))
+    >>> s
+    2000-01-01    0.000000
+    2000-01-02    0.141120
+    2000-01-03   -0.279415
+    2000-01-04    0.412118
+    2000-01-05   -0.536573
+    2000-01-06    0.650288
+    2000-01-07   -0.750987
+    2000-01-08    0.836656
+    2000-01-09   -0.905578
+    2000-01-10    0.956376
+    Freq: D, Name: A, dtype: float64
+
+    >>> s.transform(lambda x: (x - x.mean()) / x.std())
+    2000-01-01   -0.079098
+    2000-01-02    0.133921
+    2000-01-03   -0.500873
+    2000-01-04    0.542991
+    2000-01-05   -0.889050
+    2000-01-06    0.902506
+    2000-01-07   -1.212706
+    2000-01-08    1.183826
+    2000-01-09   -1.446060
+    2000-01-10    1.364543
+    Freq: D, Name: A, dtype: float64
+
+    See also
+    --------
```

Comment:
nice examples, I am not sure this will render correctly, can you post a version of this?

Reply:
They do render correctly.  
At least if the `python make.py  html --single pandas.DataFrame.transform` command generates the page which will be displayed (see screenshot below - simpler).  
(And I added one example to cover the case where the index is modified)

![screenshot-2018-3-11-pandas dataframe transform documentation](https://user-images.githubusercontent.com/2219383/37255663-cbe3e08e-254f-11e8-8640-2e53f5f312f3.png)